### PR TITLE
NiftiSpheresMasker function  _safe_get_data added

### PR DESCRIPTION
**Pull request by himanshupathak21061998**
An attempt to solve issue #1638 @KamalakerDadi @GaelVaroquaux please check this

### CODE REVIEW

**Comment by himanshupathak21061998** [comment]
File "/home/circleci/project/nilearn/input_data/nifti_spheres_masker.py", line 243, in _safe_get_data
    check_imgs = np.array(imgs.dataobj)
AttributeError: 'str' object has no attribute 'dataobj'
I can't able to understand why in circleci test imgs is treated as string please help @bthirion

**Comment by KamalakerDadi** [comment]
As per the issue, we need to implement point 1.

Point 1 requires automatically converting nans to zeros and raise a warning that we are putting nans to zeros. But, does not requires for new argument ```ensure_finite```.

At line:40 https://github.com/nilearn/nilearn/blob/master/nilearn/input_data/nifti_spheres_masker.py#L40
needs to be changed AFAIK. Instead of using niimg.get_data(), we need to use ```_safe_get_data(niimg)```. This _safe_get_data needs to be imported ```from nilearn._utils.niimg_conversions import _safe_get_data```. You don't have to implement this.

If np.isnan(np.sum(data)) then raise a warning and use np.nan_to_num to put them to zeros. This is how I do to implement point 1.

I will make a second pass when this is addressed. Then, we add tests.

**Comment by himanshupathak21061998** [comment]
Sorry for making mistakes I did not knew that utils  already contains  `_safe_get_data()`

File: nilearn/input_data/nifti_spheres_masker.py [hunk 1]
```diff
@@ -36,8 +38,17 @@ def _apply_mask_and_get_affinity(seeds, niimg, radius, allow_overlap,
...
     else:
-        mask_coords = list(np.ndindex(niimg.shape[:3]))
-        X = niimg.get_data().reshape([-1, niimg.shape[3]]).T
+        #Adding  warning message 
+        if np.isnan(np.sum(_safe_get_data(niimg))):
+            warnings.warn('The imgs you have fedded into fit_transform()' 
+                          'contains NaN values which are converted to zeroes '
+                          'please use mask_imgs to mask out NaNs othewise you '
+                           'should expect some deviation in fit_transform()')
+            mask_coords = list(np.ndindex(niimg.shape[:3]))
```
Comment (KamalakerDadi) [inline nilearn/input_data/nifti_spheres_masker.py]: This can be moved to where it was.

File: nilearn/input_data/nifti_spheres_masker.py [hunk 2]
```diff
@@ -36,8 +38,17 @@ def _apply_mask_and_get_affinity(seeds, niimg, radius, allow_overlap,
...
+            X = _safe_get_data(niimg,True).reshape([-1, niimg.shape[3]]).T
+        else:
+            mask_coords = list(np.ndindex(niimg.shape[:3]))
```
Comment (KamalakerDadi) [inline nilearn/input_data/nifti_spheres_masker.py]: You don't this.

Comment (KamalakerDadi) [reply nilearn/input_data/nifti_spheres_masker.py]: After moving this line of code outside of if or else.

File: nilearn/input_data/nifti_spheres_masker.py [hunk 3]
```diff
@@ -36,8 +38,17 @@ def _apply_mask_and_get_affinity(seeds, niimg, radius, allow_overlap,
 
         X = masking._apply_mask_fmri(niimg, mask_img)
     else:
-        mask_coords = list(np.ndindex(niimg.shape[:3]))
-        X = niimg.get_data().reshape([-1, niimg.shape[3]]).T
+        #Adding  warning message 
+        if np.isnan(np.sum(_safe_get_data(niimg))):
+            warnings.warn('The imgs you have fedded into fit_transform()' 
+                          'contains NaN values which are converted to zeroes '
+                          'please use mask_imgs to mask out NaNs othewise you '
```
Comment (KamalakerDadi) [inline nilearn/input_data/nifti_spheres_masker.py]: 'please use .....' is not needed. We can end the message at '... converted to zeros.'

File: nilearn/input_data/nifti_spheres_masker.py [hunk 4]
```diff
@@ -231,7 +242,8 @@ def __init__(self, seeds, radius=None, mask_img=None, allow_overlap=False,
         self.memory_level = memory_level
 
         self.verbose = verbose
-
+
```
Comment (KamalakerDadi) [inline nilearn/input_data/nifti_spheres_masker.py]: Please avoid touching unrelated lines.
